plugins/python-build/bin/python-build

```diff
@@ -1538,13 +1538,14 @@ use_homebrew_yaml() {
 
 use_macports_yaml() {
   can_use_macports || return 1
-  local prefix="$(port -q location libyaml 2>/dev/null || true)"
+  local port_location="$(command -v port)"
+  local prefix="${port_location%/bin/port}"
   if [ -n "$prefix" ]; then
-    local libdir="$prefix/opt/local"
-    if [ -d "$libdir" ]; then
```

Comment:
The original code `port -q location libyaml` produces `/opt/local/var/macports/software/libyaml/libyaml-0.2.5_0.darwin_24.arm64` on my machine, so `libdir` becomes `/opt/local/var/macports/software/libyaml/libyaml-0.2.5_0.darwin_24.arm64/opt/local`.

I've changed the script to detect the macport prefix, add `$prefix/include` (`/opt/local/include` on my machine) to `CPPFLAGS`, and add `$prefix/lib` to `LDFLAGS`.